### PR TITLE
Externally owned container is no longer disposed

### PR DESCRIPTION
**Pull request by danielmarbach**
Relates to https://github.com/Particular/NServiceBus/issues/4144

If a user passes us in a kernel instance we will not dispose the container instance. We consider that instance externally owned. This is a behavior breaking change. 

I also had to remove the key access since the SettingsHolder automatically disposes settings values that implement `IDisposable`.

### CODE REVIEW

File: src/NServiceBus.StructureMap.sln.DotSettings [hunk 1]
```diff
@@ -7,6 +7,7 @@
 	<s:Boolean x:Key="/Default/CodeEditing/Localization/CSharpLocalizationOptions/DontAnalyseVerbatimStrings/@EntryValue">False</s:Boolean>
 	
 	<s:Boolean x:Key="/Default/CodeInspection/ExcludedFiles/FileMasksToSkip/=_002A_002A_005C_002APackages_005C_002A_002A_005C_002A_002E_002A/@EntryIndexedValue">True</s:Boolean>
+	<s:String x:Key="/Default/CodeInspection/ExcludedFiles/FilesAndFoldersToSkip2/=3D880FCA_002DD2DA_002D4AB9_002D8017_002DBA8B7C86D359_002Fd_003AApp_005FPackages/@EntryIndexedValue">ExplicitlyExcluded</s:String>
```
Comment (WilliamBZA) [inline src/NServiceBus.StructureMap.sln.DotSettings]: :+1:

Should this be added to RepoStandards so that it pulls through when other repos are updated?

**Comment by danielmarbach** [comment]
Yep. I'll do it

File: src/NServiceBus.StructureMap.sln.DotSettings (hunk 1 above)
Comment (danielmarbach) [inline src/NServiceBus.StructureMap.sln.DotSettings]: Yep. I'll do it

Comment (danielmarbach) [inline src/NServiceBus.StructureMap.sln.DotSettings]: here we go https://github.com/Particular/RepoStandards/pull/4

**Pull request merged**